File: app/src/main/java/com/dominikgaller/alpinebooking/shared/domain/TourId.java
package com.dominikgaller.alpinebooking.shared.domain;

import java.util.Objects;

/**
 * Cross-context identity value object for an external tour definition.
 *
 * <p>Used by both the {@code booking} and {@code guide} bounded contexts
 * to reference the same tour catalog entry without coupling the contexts to each other.
 *
 * <p>Wraps a non-blank {@link String}.
 *
 * <p>SDD: See {@code documentation/adr/0003-separate-guide-bounded-context.adr.md}.
 */
public record TourId(String value) {

    public TourId {
        Objects.requireNonNull(value, "TourId value must not be null");
        if (value.isBlank()) {
            throw new IllegalArgumentException("TourId value must not be blank");
        }
    }
}

File: src/main/kotlin/com/example/contractmanagement/shared/domain/TourId.kt
package com.example.contractmanagement.shared.domain

/**
 * Cross-context identity value object for an external tour definition.
 *
 * Used by both the `booking` and `guide` bounded contexts to reference the same tour
 * catalog entry without coupling the contexts to each other.
 *
 * SDD: see `documentation/adr/0003-separate-guide-bounded-context.adr.md`.
 */
data class TourId(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "TourId value must not be blank" }
    }
}
